src/datamigration/azext_datamigration/manual/helper.py

```diff
@@ -460,8 +500,35 @@ def is_user_admin():
 # Helper function to validate key input.
 # -----------------------------------------------------------------------------------------------------------------
 def validate_input(key):
-    if key == "":
-        raise InvalidArgumentValueError("Failed: IR Auth key is empty. Please provide a valid auth key.")
+
+    if is_string_null_or_empty(key):
+        raise InvalidArgumentValueError("Failed: IR Auth key is null or empty. Please provide a valid auth key.")
+    if has_shell_escape_characters(key):
+        raise InvalidArgumentValueError("Failed: IR Auth key has invalid characters. Please provide a valid auth key.")
+    if not is_valid_ir_key_format(key):
+        raise InvalidArgumentValueError("Failed: Invalid IR Auth key format. Please provide a valid auth key.")
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check if the input key is in valid format.
+# -----------------------------------------------------------------------------------------------------------------
+def is_valid_ir_key_format(key):
+
```

Comment:
Few things for readability
- mention the IR format in comments like "IR@<>@<>@..."
- If required, you can store each part like
guid = key_parts[1] and then perform the check.
- since there are multiple parts and the same check, you can consider having a for loop
    for parts in key_parts[1:]:
          if is_string_null_or_empty(parts):
              return False
- move the key parts length into a separate variable, can keep it as a local variable.

Reply:
- maybe key.startswith(keyprefix) instead of key_part[0] != keyPrefix
- For point 3, lets just loop over all parts for null_or_empty check

```
for key_part in key_parts:
    if is_string_null_or_empty(key_part):
        return False
```
 
- Instead of hardcoding key_part[4] maybe we can access last part as key_part[-1]

Reply:
Added the commented changes.